internal/tofu/transform_reference.go

```diff
@@ -454,10 +461,46 @@ func (m ReferenceMap) dataDependsOn(depender graphNodeDependsOn) []*addrs.Refere
 	return refs
 }
 
-// parentModuleDependsOn returns the set of vertices that a data sources parent
+// ephemeralDependsOn returns extra depends_on references if this is an ephemeral resource.
+// For ephemeral resources we implicitly treat references to managed resources as
+// depends_on entries. If an ephemeral resource references a managed resource, even if
+// that reference is resolvable, it stands to reason that the user intends for
+// the ephemeral resource to require that resource in some way.
+func (m ReferenceMap) ephemeralDependsOn(depender graphNodeDependsOn) []*addrs.Reference {
```

Comment:
As far as I can tell, this is nearly identical to the above function, is there any reason to split them out?

Reply:
I did it this way just to split the logic, but yeah, at a second check on this, I moved the splitting one level below so I removed this one.